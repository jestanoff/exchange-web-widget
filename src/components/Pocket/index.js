import React from 'react';
import { oneOf } from 'prop-types';
import classNames from 'classnames';
import Input from '../Forms/Input';
import Balance from '../Balance';
import getCurrencyCodes from '../../utils/getCurrencyCodes';
import * as currencies from '../../constants/currencies';
import styles from './pocket.css';

const FromPocket = ({ currency, type }) => (
  <section className={classNames(styles.container, styles[type])}>
    <div>
      <h2 className={styles.currency}>{currency}</h2>
      <Balance currency={currency} />
    </div>
    <Input name={type} />
  </section>
);

FromPocket.propTypes = {
  currency: oneOf(getCurrencyCodes(currencies)).isRequired,
  type: oneOf(['from', 'to']).isRequired,
};

export default FromPocket;
