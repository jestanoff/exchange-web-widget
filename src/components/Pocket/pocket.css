.container {
  align-items: start;
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  padding: 24px 16px;
}

.to {
  background: #F0F0F0;
}

.from {
  background: #FFFFFF;
}

.currency {
  font-size: 36px;
  margin-bottom: 6px;
}
