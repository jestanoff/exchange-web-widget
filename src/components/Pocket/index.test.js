
import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Pocket from './';


describe('Pocket component', () => {
  test('should render proper markup', () => {
    const { container } = render(<Pocket currency="EUR" type="to" />);
  
    expect(container).toMatchSnapshot();
  })
});
