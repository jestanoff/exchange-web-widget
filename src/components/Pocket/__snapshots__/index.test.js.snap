// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pocket component should render proper markup 1`] = `
<div>
  <section
    class="container to"
  >
    <div>
      <h2
        class="currency"
      >
        EUR
      </h2>
      <span
        class="balance"
      >
        Balance: 
        â‚¬0
      </span>
    </div>
    <input
      class="input"
      data-test="input-to"
      maxlength="17"
      name="to"
      placeholder="0"
      required=""
      type="tel"
    />
  </section>
</div>
`;
